torch/_subclasses/functional_tensor.py

```diff
@@ -537,6 +559,29 @@ def unwrap(x):
                         torch.Tensor, wrap, outs_unwrapped
                     )
                 else:
+                    arg_mutation_counters = []
+
+                    if self.export:
```

Comment:
another nit: I'd like to keep the `FunctionalTensorMode.__torch_dispatch__` code as readable as possible, and I'm a bit worried about it being dominated by `is_export` branching code. A few things I think will help:

(1) instead of a separate branch on `self.export` both before and after redispatching, you can probably just have all of your checks run after the redispatch. Even if we are supposed to error because we are tracing some code that passes a frozen tensor into an op, there is nothing wrong with executing that op under tracing first before raising the error

(2) let's move most of the logic for manipulating those private dictionaries into a helper method